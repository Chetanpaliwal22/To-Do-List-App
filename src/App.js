import React from 'react';
import tododatas from './tododatas'
import TodoItem from './ToDoItem'
import Button from 'react-bootstrap/Button'
import 'bootstrap/dist/css/bootstrap.min.css'

class App extends React.Component {
  constructor() {
    super()
    this.state = {
      todos: tododatas,
    }

    this.handleChange = this.handleChange.bind(this)
    this.addTask = this.addTask.bind(this);
    this.deleteTask = this.deleteTask.bind(this);
  }

  handleChange(id) {
    
    this.setState(prevState => {
      
      let updatedTodos = prevState.todos.map(todo => {
        //console.log(todo.completed)
        if (todo.id === id) {
          todo.completed = !todo.completed
        }
        return todo
      })
      
      return {
        todos: updatedTodos
      }
    })
  }

  render() {
    const todoItems = this.state.todos.map(item => <TodoItem key={item.id} item={item} handleChange={this.handleChange} />)

    return (
      <div className="todo-list">
        <div >
          <h1> TO-DO LIST </h1>
          <h3>Hey Chetan, what's on your mind today?</h3>
          {todoItems}
        </div>

        <div className="todo-button">
          <Button variant="outline-success" onClick={this.addTask}>Add Task</Button>
          <Button variant="outline-danger" onClick={this.deleteTask}>Delete Task</Button>
        </div>
      </div>
    )
  }

  addTask() {
    let len = tododatas.length
    tododatas.push({
      id: len + 1,
      text: "new task push the code to git and update readme",
      completed: false
    })
    this.setState({
      todos: tododatas
    })
  }

  deleteTask() {
    tododatas.pop();
    this.setState({
      todos: tododatas
    })
  }

}
export default App;
