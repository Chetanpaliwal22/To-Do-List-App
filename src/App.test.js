import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';

test('Snapshot for App component', () => {

  const renderedComponent = renderer.create(<App/>).toJSON();
  expect(renderedComponent).toMatchSnapshot();
});
