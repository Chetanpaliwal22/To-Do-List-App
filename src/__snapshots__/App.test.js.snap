// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for App component 1`] = `
<div
  className="todo-list"
>
  <div>
    <h1>
       TO-DO LIST 
    </h1>
    <h3>
      Hey Chetan, what's on your mind today?
    </h3>
  </div>
  <div
    className="todo-button"
  >
    <button
      className="btn btn-outline-success"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Add Task
    </button>
    <button
      className="btn btn-outline-danger"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Delete Task
    </button>
  </div>
</div>
`;
